packages/browser/__tests__/login/oidc/redirectHandler/ErrorOidcHandler.spec.ts

```diff
@@ -0,0 +1,83 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, it, describe, expect } from "@jest/globals";
+import { ErrorOidcHandler } from "../../../../src/login/oidc/redirectHandler/ErrorOidcHandler";
+
+describe("ErrorOidcHandler", () => {
+  describe("canHandle", () => {
+    it("always accept the given IRI", async () => {
+      const redirectHandler = new ErrorOidcHandler();
+      expect(
+        await redirectHandler.canHandle(
+          "https://coolparty.com/?code=someCode&state=oauth2_state_value"
+        )
+      ).toBe(false);
+      expect(await redirectHandler.canHandle("https://coolparty.com/")).toBe(
+        false
+      );
+      expect(
+        await redirectHandler.canHandle("https://coolparty.com/?test=test")
+      ).toBe(false);
+    });
+
+    it("handles url with error params", async () => {
+      const redirectHandler = new ErrorOidcHandler();
+      expect(
+        await redirectHandler.canHandle(
+          "https://coolparty.com/?error=some_error"
+        )
+      ).toBe(true);
+    });
+
+    it("throws on invalid url", async () => {
+      const redirectHandler = new ErrorOidcHandler();
+      await expect(
+        redirectHandler.canHandle("beep boop I am a robot")
+      ).rejects.toThrow(
+        "[beep boop I am a robot] is not a valid URL, and cannot be used as a redirect URL: TypeError: Invalid URL: beep boop I am a robot"
+      );
+    });
+  });
+
+  describe("handle", () => {
+    it("returns an unauthenticated session", async () => {
+      window.fetch = jest.fn();
+      const redirectHandler = new ErrorOidcHandler();
+      const mySession = await redirectHandler.handle("https://my.app");
+      expect(mySession.isLoggedIn).toEqual(false);
+      expect(mySession.webId).toBeUndefined();
+    });
+    it("calls on Error if given", async () => {
+      window.fetch = jest.fn();
+      const redirectHandler = new ErrorOidcHandler();
+      const mockCallback = jest.fn();
+      const mySession = await redirectHandler.handle(
+        "https://coolparty.com/?error=fetchError&errorDescription=unable_to_fetch",
+        undefined,
+        mockCallback
+      );
+      expect(mySession.isLoggedIn).toEqual(false);
+      expect(mySession.webId).toBeUndefined();
+      expect(mockCallback.mock.calls).toHaveLength(1);
```

Comment:
Ideally, this would also test that the callback has been called with the appropriate parameters, i.e. doing something like `      expect(mockCallback).toHaveBeenCalledWith("error", "unable_to_fetch");`